File: src/ProjectEuler/Problems/Problem1.hs
-- 233168 - Completed 29.4.2013
module ProjectEuler.Problems.Problem1
    ( problem1  -- :: IO ()
    , problem1' -- :: IO ()
    ) where

import Data.List (union)

problem1 :: IO ()
problem1 = print $ sum [x | x <- [1..999], x `rem` 3 == 0 || x `rem` 5 == 0]

problem1' :: IO ()
problem1' = print $ sum $ union [3,6..999] [5,10..995]

File: src/ProjectEuler/C/Problems/Problem1.c
// 233168 - Completed 20.8.2013
#include <stdio.h>

int main(){
    int sum = 0,n;
    for (0; n < 1000; n++){
	sum = (n % 3 == 0 || n % 5 == 0) ? sum + n : sum;
    }
    printf("%d\n",sum);
    return 0;
}
